core/src/main/scala/org/apache/spark/sql/execution/row/RowFormatScanRDD.scala

```diff
@@ -411,7 +414,7 @@ final class InternalRowIteratorOnRS(conn: Connection,
       i = i + 1
     }
     moveNext()
-    mutableRow
+    unsafeproj.apply(mutableRow)
```

Comment:
Above does two copies, once into SpecificMutableRow then to UnsafeRow which can be optimized (+ avoiding UTF8 conversion by using same format in GemXD as mentioned in TODO). Can be low priority since it is for row tables only which we will deprecate in long term.